### PR TITLE
Lightning interface for the core readout model

### PR DESCRIPTION
**Pull request by thomasZen**
- Simplify Readout layer
- Add lightning interface to the core readout model
- Use config for training

## Training on the cluster
Run the following command from any directory:
```
~/GitRepos/open-retina/scripts/train_core_readout.py \
--config-path ~/GitRepos/open-retina/configs \
--config-name example_train_core_readout
```
The current scores with this approach are (0=train, 1=validation, 2=test):
```
────────────────────────────────────────────────────────────────────────────────────────────────────────────────
       Test metric             DataLoader 0             DataLoader 1             DataLoader 2                                                                                                               
────────────────────────────────────────────────────────────────────────────────────────────────────────────────
    test_correlation        0.27609729766845703      0.3052205443382263       0.5100897550582886                                                                                                            
        test_loss            817.4778442382812        627.6156005859375        82.57406616210938                                                                                                            
────────────────────────────────────────────────────────────────────────────────────────────────────────────────
```

## Config
I used [hydra](https://hydra.cc/docs/intro/) for config parsing. This has a few advantages:
- You can dynamically instantiate objects, see loggers in example_config
- You can overwrite any config parameter from the command line, see [hydra docs](https://hydra.cc/docs/advanced/override_grammar/basic/)

## Todo
- Spatial Laplace is not implemented

### CODE REVIEW

File: example_configs/train_core_readout.yaml
Comment (fededagos) [inline example_configs/train_core_readout.yaml:1]: Can you maybe name the configs folder just "configs" and this config "example_train_core_readout.yaml"? This way in the future we can add in sensible default configs for hydra to use, also taking advantage of the fact that we can nest and compose them using hydra.

Comment (thomasZen) [reply example_configs/train_core_readout.yaml:1]: Makes sense, I changed it.

**Review by fededagos — APPROVED** [review]
All clear! Will need to experiment with this a bit, I will ask you questions as they arise :)

**Comment by thomasZen** [comment]
> All clear! Will need to experiment with this a bit, I will ask you questions as they arise :)

Perfect, I'll go ahead and merge then. We can also improve things in the future then :)

File: scripts/train_core_readout.py [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+    data_folder = os.path.expanduser(conf.data_folder)
+    movies_path = os.path.join(data_folder, "2024-01-11_movies_dict_8c18928.pkl")
+    with open(movies_path, "rb") as f:
+        movies_dict = pickle.load(f)
+
+    data_path_responses = os.path.join(data_folder, "2024-03-28_neuron_data_responses_484c12d_djimaging.h5")
+    responses = load_h5_into_dict(data_path_responses)
+
+    data_dict = make_final_responses(responses, response_type="natural")  # type: ignore
+    dataloaders = natmov_dataloaders_v2(data_dict, movies_dictionary=movies_dict, train_chunk_size=100, seed=1000)
```
Comment (thomasZen) [inline scripts/train_core_readout.py:31]: Here's were we wrote train_chunk_size=100

**Pull request merged**